src/connection.js

```diff
@@ -98,86 +96,78 @@ class Connection extends EventEmitter {
         if (options.password !== undefined && typeof options.password !== 'string') {
           throw new TypeError('The "config.authentication.options.password" property must be of type string.');
         }
-      }
 
-      authentication = { type: authType };
-      switch (authentication.type) {
```

Comment:
Hi @arthurschreiber, I got this part form Hadis. Before she left, she did working on the MSI authentication for a while, and created this switch statement instead of set the similar object for every authentication types using multiple levels of if and else.  In my option, I think when we only have three authentication types, it makes sense that we only use if and else to handle different cases. However, now there are more types, and some of them do share some common config.authentication.options setups, I think the switch statement may actually help. Saved us some readily used if statements, and config initialization code.

Reply:
Hey @MichaelSun90 👋 

I understand where you where going with this, but there were some issues with the changes for this (like some validations going missing) and the diff for the PR was becoming hard to review because these changes are a bit unrelated to the new functionality.

I think we can and should tackle this again in the future, but not in this PR. 🙇

Reply:
Hi @arthurschreiber, Thanks for making all the changes recently. :thumbsup:  It makes sense to me to separate this into a future PR since it indeed not closely related to the MSI authentication feature itself.